deployments/databricks/deploy_collector.sh

```diff
@@ -0,0 +1,170 @@
+#   https://docs.databricks.com/en/integrations/compute-details.html
+# - DATABRICKS_ACCESS_TOKEN: Databricks personal access token (PAT) used to connect to the Apache Spark API.
+#   Directions for creating a PAT: https://docs.databricks.com/en/dev-tools/auth/pat.html
+
+# Optional Variables:
+# - OTEL_VERSION: Version of the Splunk OpenTelemetry Collector to deploy as a part of this release.
+#   Default: "latest". Valid version must be >=0.120.0.
+# - SCRIPT_DIR: Installation path for the Collector and its config
+# - SPLUNK_REALM: Splunk o11y realm to send data to. Default: us0
+
+DATABRICKS_CLUSTER_HOSTNAME=${DATABRICKS_CLUSTER_HOSTNAME:-}
+SPLUNK_ACCESS_TOKEN=${SPLUNK_ACCESS_TOKEN:-}
+DATABRICKS_ACCESS_TOKEN=${DATABRICKS_ACCESS_TOKEN:-}
+
+if [ -z "${DATABRICKS_CLUSTER_HOSTNAME}" ]; then
+  echo "environment variable 'DATABRICKS_CLUSTER_HOSTNAME' must be set, exiting."
+  exit 1
+fi
+
+if [ -z "${SPLUNK_ACCESS_TOKEN}" ]; then
+  echo "environment variable 'SPLUNK_ACCESS_TOKEN' must be set, exiting."
+  exit 1
+fi
+
+if [ -z "${SDATABRICKS_ACCESS_TOKEN}" ]; then
+  echo "environment variable 'DATABRICKS_ACCESS_TOKEN' must be set, exiting."
+fi
+
+
+OTEL_VERSION=${OTEL_VERSION:-latest}
+OS="linux_amd64"
+OTEL_BINARY_NAME="splunk_otel_collector"
+OTEL_DOWNLOAD_BASE_URL="https://github.com/signalfx/splunk-otel-collector/releases"
+OTEL_API_URL="https://api.github.com/repos/signalfx/splunk-otel-collector/releases/latest"
+SCRIPT_DIR=${SCRIPT_DIR:-/tmp/collector_download}
+CONFIG_FILENAME="config.yaml"
+OTEL_BINARY_FILE="$SCRIPT_DIR/$OTEL_BINARY_NAME"
+CONFIG_FILE="$SCRIPT_DIR/$CONFIG_FILENAME"
+SERVICE_PATH="/etc/systemd/system/"
+SERVICE_FILE="$SERVICE_PATH/$OTEL_BINARY_NAME.service"
+
+if [ $OTEL_VERSION = "latest" ]; then
+        OTEL_VERSION=$(curl --silent "$OTEL_API_URL" |    # Get latest Collector release from GitHub api
+            grep '"tag_name":' |                          # Get tag name line
+            sed -E 's/.*"([^"]+)".*/\1/')                 # Pluck latest release version
+        if [ -z "$OTEL_VERSION" ]; then
+            echo "Failed to get tag_name for latest release from $OTEL_VERSION/latest" >&2
+            exit 1
+        fi
+fi
+
+OTEL_BINARY_DOWNLOAD_URL=${OTEL_DOWNLOAD_BASE_URL}/download/${OTEL_VERSION}/otelcol_${OS}
+
+# Download Splunk's distribution of the OpenTelemetry Collector
+mkdir -p "$SCRIPT_DIR"
+wget -nv -O "$OTEL_BINARY_FILE" $OTEL_BINARY_DOWNLOAD_URL
+chmod +x "$OTEL_BINARY_FILE"
+
+# The Spark receiver should only be run in one instance per-Cluster. Run
+# it on the driver node, as there's one per-cluster.
+# More info on Databricks init script environment variables:
+# https://docs.databricks.com/en/init-scripts/environment-variables.html#use-secrets-in-init-scripts
+if [ $DB_IS_DRIVER = "TRUE" ]; then
+        OPTIONAL_SPARK_RECEIVER=", apachespark"
+else
+        OPTIONAL_SPARK_RECEIVER=""
+fi
+
+collector_config="
+extensions:
+  bearertokenauth:
+    token: "$DATABRICKS_ACCESS_TOKEN"
```

Comment:
```suggestion
    token: ${env:DATABRICKS_ACCESS_TOKEN}
```
Should we use env config provider instead? It would help keep the config file free of sensitive tokens. Also, I think we might have to escape the $ in my suggestion.

Reply:
I agree with your concern here, but after some research, the collector is run by systemd which doesn't have access to environment variables. The collector fails to start because the environment variable can't be found.

The alternative is to add the access token as an [environment variable to the service](https://serverfault.com/questions/413397/how-to-set-environment-variable-in-systemd-service), but I think that ends up with a similar problem where the service script now contains the plaintext access token.

Any other ideas are welcome